src/YagrCore/plugins/legend/legend.ts

```diff
@@ -333,26 +333,31 @@ export default class Legend {
     }
 
     private renderItems(uplotOptions: Options) {
-        const title = getPrependingTitle(this.yagr.i18n, uplotOptions.series);
+        type SeriesRenderData = {
+            series: string | Series;
+            idx: number;
+        };
```

Comment:
Is it ok to define a type inside a method?

Reply:
Agreed, it is not generic so maybe it would be better to move it to the top of the file

Reply:
It using only in this scope